### ISSUE TITLE
Mock with Federations - Syntax Error: Unexpected Name "Hello"

### ISSUE THREAD

**Issue by vmptk** [apollographql/federation#348]
When using the '@apollo/federation' and ''@apollo/gateway' with mocked types and with micro server ('apollo-server-micro'). An error occurs on the gateway service:
```
Encountered error when loading images at http://localhost:4003/graphql: Syntax Error: Unexpected Name "Hello"
```

Expected behaviour is to properly build the graph and use mocked types.
Actual behaviour, gateway is failing to load the types.

A workaround is to set the option:     mockEntireSchema: false,

**Comment by nemisj** [comment]
> > 1. Find a way to return the SDL even when mock: true is passed.
> 
> I've been playing with the code and I've found a way to have Federation to work with a federated service using mock. Here is what you would need to do:
> 
> ```
> // First declare your schema as a separate const
> const schemaString = `
>   type User {
>     _id: ID!
>   }
> 
>   extend type Query {
>     user(id: ID!): User
>   }
> `;
> 
> // Create type def of your string schema using the gql
> const typeDefs = gql(schemaString);
> 
> // Create mock of the _Service type which will be used by the Apollo Gateway to fetch your schema definition
> const mocks = {
>   _Service: () => ({
>     'sdl': schemaString
>   }),
> };
> 
> const options = {
>   typeDefs,
>   resolvers,
>   mocks,
> };
> 
> const server = new ApolloServer(options);
> ```
> 
> > 1. Use Apollo Graph Manager.
> 
> Tomorrow I will attempt to use Apollo Graph Manager instead of passing a `serviceList`. It feels a cleaner solution to me.
> 
> In the meantime, I hope this temporary solution will unblock you all :)

There is also string based query available on typeDefs, so that you do not need declare your schema as separate variable:
```
const mocks = {
  _Service: () => ({
    'sdl': typeDefs?.loc?.source?.body
  }),
};
```

**Comment by kevin-lindsay-1** [comment]
I had this issue recently, just set the following on the federated service:
```ts
new ApolloServer({
  ...,
  mockEntireSchema: false,
  mocks: {},
})
```

And had the gateway use as normal.

**Comment by namirsab** [comment]
Hey!
This issue is still opened, so I guess there is no real solution for it? Are there any news?

**Comment by trevor-scheer** [comment]
Closed via #1773

https://github.com/apollographql/federation/pull/1773#issuecomment-1109014778
